src/normalizeOptions.js

```diff
@@ -51,29 +51,42 @@ function normalizeRoot(opts) {
   }
 }
 
-function normalizeAlias(opts) {
-  opts.regExps = [];
-
-  if (opts.alias) {
-    Object.keys(opts.alias)
-      .filter(isRegExp)
-      .forEach((key) => {
-        const parts = opts.alias[key].split('\\\\');
-
-        function substitute(execResult) {
-          return parts
-            .map(part =>
-              part.replace(/\\\d+/g, number => execResult[number.slice(1)] || ''),
-            )
-            .join('\\');
-        }
+function getAliasPair(key, value) {
+  const parts = value.split('\\\\');
+
+  function substitute(execResult) {
+    return parts
+      .map(part =>
+        part.replace(/\\\d+/g, number => execResult[number.slice(1)] || ''),
+      )
+      .join('\\');
+  }
 
-        opts.regExps.push([new RegExp(key), substitute]);
+  return [new RegExp(key), substitute];
+}
 
-        delete opts.alias[key];
-      });
+function normalizeAlias(opts) {
+  if (opts.alias) {
+    const { alias } = opts;
+    const aliasKeys = Object.keys(alias);
+    const nonRegExpAliases = [];
+    const regExpAliases = [];
+
+    aliasKeys.forEach((key) => {
+      if (isRegExp(key)) {
+        regExpAliases.push(
+          getAliasPair(key, alias[key]),
+        );
+      } else {
+        nonRegExpAliases.push(
+          getAliasPair(`^${key}((?:/|).*)`, `${alias[key]}\\1`),
+        );
+      }
+    });
+
+    opts.alias = [...nonRegExpAliases, ...regExpAliases];
```

Comment:
I wanted to preserve the order: standard aliases before regexps. At this point I think it could go either way, as the order should be taken from the object.

Note: according to the spec, the order of enumerating the object keys is not guaranteed to be the same as in the literal, so in the future it may be needed to allow an array-type declaration. We're currently relying on V8.